lib/Dialect/HW/HWOps.cpp

```diff
@@ -352,15 +353,20 @@ static void buildModule(OpBuilder &builder, OperationState &result,
   result.addAttribute(mlir::function_like_impl::getResultDictAttrName(),
                       builder.getArrayAttr(resultAttrs));
   result.addAttribute("parameters", parameters);
+  if (comment)
+    result.addAttribute("comment", comment);
+  else
+    result.addAttribute("comment", builder.getStringAttr(""));
```

Comment:
Maybe simplify this to `if (!comment) comment =  builder.getStringAttr("");`?

Reply:
Nice. Switched to this in the rebase.